backend/user-service/config/config.go

```diff
@@ -59,12 +60,14 @@ func LoadConfig(path string) (config Config, err error) {
 
 	err = viper.ReadInConfig()
 	if err != nil {
-		return
+		return Config{}, err
 	}
 
-	err = viper.Unmarshal(&config)
+	err = viper.Unmarshal(&config, func(dc *mapstructure.DecoderConfig) {
+		dc.ErrorUnset = true
+	})
 	if err != nil {
-		return
+		return Config{}, err
```

Comment:
You should use *config instead to return nil, err.

Passing a big struct and copying around the program is not good.